nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,434 @@
+
+        self.logger.debug(f"Status at nipoppy_checkpoint: {nipoppy_checkpoint}")
+        doughnut = self.doughnut
+        doughnut_cols = [
+            doughnut.col_in_pre_reorg,
+            doughnut.col_in_post_reorg,
+            doughnut.col_in_bids,
+        ]
+
+        # Get the number of participants in the doughnut
+        participant_ids = doughnut[doughnut.col_participant_id].unique()
+        session_ids = doughnut[doughnut.col_session_id].unique()
+
+        self.logger.debug(
+            f"\tNumber of participants in doughnut: {len(participant_ids)}"
+        )
+        self.logger.debug(f"\tAvailable sessions (n={len(session_ids)}): {session_ids}")
+
+        doughnut_status_df = doughnut.groupby([doughnut.col_session_id]).sum()[
+            doughnut_cols
+        ]
+
+        self.logger.debug(f"doughnut_status_df: {doughnut_status_df}")
+        status_df = pd.concat([status_df, doughnut_status_df], axis=1)
+        return status_df, doughnut_cols
+
+    def _check_bagel(self, status_df: pd.DataFrame) -> pd.DataFrame:
+        """Check the imaging bagel file (if exists)."""
+        nipoppy_checkpoint = "in_imaging_bagel"
+
+        self.logger.debug(f"Status at nipoppy_checkpoint: {nipoppy_checkpoint}")
+        bagel = self.bagel
+
+        # Get the number of participants in the bagel
+        participant_ids = bagel[bagel.col_participant_id].unique()
+        session_ids = bagel[bagel.col_session_id].unique()
+        pipelines = bagel[bagel.col_pipeline_name].unique()
+
+        self.logger.debug(f"\tNumber of participants in bagel: {len(participant_ids)}")
+        self.logger.debug(f"\tAvailable visits (n={len(session_ids)}): {session_ids}")
+        self.logger.debug(f"\tAvailable pipelines (n={len(pipelines)}): {pipelines}")
+
+        bagel[self.col_pipeline] = (
+            bagel[bagel.col_pipeline_name]
+            + "\n"
+            + bagel[bagel.col_pipeline_version]
+            + "\n"
+            + bagel[bagel.col_pipeline_step]
+        )
+
+        bagel_pipeline_df = bagel[
+            [
+                bagel.col_participant_id,
+                bagel.col_session_id,
+                self.col_pipeline,
+                bagel.col_status,
+            ]
+        ]
+        bagel_pipeline_df = bagel_pipeline_df[
+            bagel_pipeline_df[bagel.col_status] == STATUS_SUCCESS
+        ]
+
+        bagel_pipeline_df = bagel_pipeline_df.pivot(
+            index=[bagel.col_participant_id, bagel.col_session_id],
+            columns=self.col_pipeline,
+            values=bagel.col_status,
+        )
+
+        bagel_status_df = bagel_pipeline_df.groupby([bagel.col_session_id]).count()
+
+        self.logger.debug(f"bagel_status_df: {bagel_status_df}")
+
+        status_df = pd.concat([status_df, bagel_status_df], axis=1)
+
+        bagel_cols = list(status_df.columns[status_df.columns.str.contains("\n")])
+        return status_df, bagel_cols
+
+    def _add_manifest_rewards(
```

Comment:
It might be good to have some comments explaining the `if`/`elif`/`else` conditions. For all three reward conditions